### PR TITLE
fix(secret): scan secrets in github repositories

### PR DESCRIPTION
**Pull request by cajlopezor**
## Description

Add code to secret scan on GitHub Repositories 

### Fix
*How does someone fix the issue in code and/or in runtime?* Yes

## Checklist:

- [x] The pull request is complete according to the guide of [contributing](https://github.com/bancolombia/devsecops-engine-tools/blob/trunk/docs/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: tools/devsecops_engine_tools/engine_sast/engine_secret/src/domain/usecases/secret_scan.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def process(self, skip_tool, config_tool):
         if skip_tool == False:
             self.tool_gateway.install_tool(self.devops_platform_gateway.get_variable("os"), self.devops_platform_gateway.get_variable("temp_directory"))
             files_pullrequest = self.git_gateway.get_files_pull_request(
-                self.devops_platform_gateway.get_variable("work_folder"),
```
Comment (Gaviria9601) [inline tools/devsecops_engine_tools/engine_sast/engine_secret/src/domain/usecases/secret_scan.py:37]: If this variable is not used , please remove the implementation in the github,azure devops and runtime_local driven adapters. Also the referefenre of DET_WORK_FOLDER.

File: tools/devsecops_engine_tools/version.py [hunk 2]
```diff
@@ -1 +1 @@
-version = '1.7.41'
+version = '1.7.42'
```
Comment (Gaviria9601) [inline tools/devsecops_engine_tools/version.py]: Update the version, as it has conflicts

File: tools/devsecops_engine_tools/engine_sast/engine_secret/src/domain/usecases/secret_scan.py (hunk 1 above)
Comment (cajlopezor) [reply tools/devsecops_engine_tools/engine_sast/engine_secret/src/domain/usecases/secret_scan.py:37]: delete variable work_folder

File: tools/devsecops_engine_tools/version.py (hunk 2 above)
Comment (cajlopezor) [reply tools/devsecops_engine_tools/version.py]: update to version 1.8.1

**Review by Gaviria9601 — COMMENTED** [review]
Please. Remove these references
image

**Review by Gaviria9601 — APPROVED** [review]

**Pull request merged**